src/commcare_cloud/ansible/roles/datadog/tasks/main.yml

```diff
@@ -20,6 +20,23 @@
     repo: 'deb http://apt.datadoghq.com/ stable main'
     state: absent
 
+# the GPG keys were not getting cleaned up by the apt_key module, so we need to do it manually
+- name: Remove all old datadog key files
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - /etc/apt/keyrings/DATADOG_APT_KEY_382E94DE.gpg
+    - /etc/apt/keyrings/DATADOG_APT_KEY_F14F620E.gpg
+    - /etc/apt/keyrings/DATADOG_APT_KEY_C0962C7D.gpg
```

Comment:
@AmitPhulera in hindsight, `DATADOG_APT_KEY_C0962C7D.gpg` is not yet an old key right? It gets added back in "Add new datadog apt key" below. Was this intentional?

Reply:
Yes. This was intentional. I think the task should be called remove all existing keys instead of old keys. We are doing it to start from a clean slate.
And if I remember correctly not removing that key added same key twice in soucrce list which again caused issues.